storyscript/hub/sdk/ServiceWrapper.py

```diff
@@ -0,0 +1,132 @@
+import json
+from uuid import UUID
+
+from storyscript.hub.sdk.GraphQL import GraphQL
+
+from storyscript.hub.sdk.service.ServiceData import ServiceData
+
+
+class UUIDEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, UUID):
+            return obj.hex
+        return json.JSONEncoder.default(self, obj)
+
+
+class ServiceWrapper:
+    """
+    The ServiceWrapper provides an improved way to access storyscript hub services
+    """
+    def __init__(self, services=None):
+        self.services = {}
+        self.reload_services(services)
+
+    @classmethod
+    def from_dict(cls, dictionary=None):
+        services = []
+
+        if dictionary is not None:
+            services = dictionary
+
+        return cls(services)
+
+    @classmethod
+    def from_json(cls, jsonstr=None):
+        services = []
+
+        if jsonstr is not None:
+            services = json.loads(jsonstr)
+
+        return cls(services)
+
+    @classmethod
+    def from_json_file(cls, path=None):
+        if path is not None:
+            with open(path, 'r') as f:
+                jsonstr = f.read()
+                f.close()
+                return cls.from_json(jsonstr=jsonstr)
+
+        return cls([])
+
+    def reload_services(self, services=None):
+        # reset services
+        self.services = {}
+
+        all_services = None
+        if type(services) is list:
+            for service in services:
+                if type(service) is dict:
+                    service_data = service["service"]
+                    self.services[(service_data["owner"]["username"] + '/' + service_data["name"])] = service
+                elif type(service) is str:
+                    # this allows us to utilize dynamic loading
+                    if not all_services:
+                        all_services = GraphQL.get_all()
+
+                    for _service in all_services:
+                        service_owner = _service["service"]["owner"]["username"]
+                        service_name = _service["service"]["name"]
+                        service_alias = _service["service"]["alias"]
+                        if service == service_owner + "/" + service_name or service == service_alias:
+                            self.services[service_owner + "/" + service_name] = _service
+
+        elif type(services) is dict:
+            for service in services:
+                _service = services[service]
+                service_owner = _service["service"]["owner"]["username"]
+                service_name = _service["service"]["name"]
+                self.services[service_owner + "/" + service_name] = _service
+
+    def as_json(self):
+        services = []
+
+        for service in self.services:
+            services.append(self.services[service])
+
+        return json.dumps(services, indent=4, sort_keys=True, cls=UUIDEncoder)
+
+    def as_json_file(self, out_file=None):
```

Comment:
```suggestion
    def as_json_file(self, out_file):
```

This isn't `optional`